flocker/acceptance/testtools.py

```diff
@@ -547,13 +572,14 @@ def created():
             def got_results(results):
                 # State has unpredictable path, so we don't bother
                 # checking for its contents:
-                actual_dataset_states = [d.set(path=None) for d in results]
-                return expected_dataset_state in actual_dataset_states
+                actual_dataset_states = list(
+                    d for d in results
+                    if d.set('path', None) == expected_dataset_state)
+                return (actual_dataset_states or [None])[0]
```

Comment:
:question: I may just be old and slow but I find these expressions difficult to read. Personally I'd rather see:
```
if matching_dataset_states:
    [matching_dataset_state] = matching_dataset_states
    return matching_dataset_state
else:
    return None
```

(or something similar)